### PR TITLE
feat: Allow the writing of Json object to Cbor

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
This PR adds the option to write the material map to a Cbor file instead of a Json one.  Doing so reduce the size of the file by a factor of around 10 but make the file non longer human-readable. This should be used after the map has been optimised when we want to export it.

Also fix an error related to the material mapping python config script.

### CODE REVIEW

File: Examples/Io/Json/include/ActsExamples/Io/Json/JsonMaterialWriter.hpp [hunk 1]
```diff
@@ -39,11 +39,21 @@ namespace ActsExamples {
...
  public:
+  struct Config {
+    /// The config class of the converter
+    Acts::MaterialMapJsonConverter::Config converterCfg;
+    /// Output file name
+    std::string fileName = "material";
+    /// Write the material map as a Json file
+    bool writeJson = true;
+    /// Write the material map as a CBOR file
+    bool writeCbor = false;
```
Comment (paulgessinger) [inline Examples/Io/Json/include/ActsExamples/Io/Json/JsonMaterialWriter.hpp:50]: Just a suggestion, but how about making this an enum? That way, we could add support other JSON-like formats if we want to later on. We do have a similar enum in the Examples actually.

File: docs/howto/run_material_mapping.rst [hunk 2]
```diff
@@ -15,7 +15,7 @@ First we will need to extract the list of all the surfaces and volumes in our de
 
 .. code-block:: console
 
-  ./../build/bin/ActsExampleGeometryDD4hep -n1 -j1 --mat-output-file geometry-map  --dd4hep-input ../Examples/Detectors/DD4hepDetector/compact/OpenDataDetector/OpenDataDetector.xml --output-json --mat-output-allmaterial true --mat-output-sensitives false
+  ./../build/bin/ActsExampleGeometryDD4hep -n1 -j1 --mat-output-file geometry-map  --dd4hep-input ../thirdparty/OpenDataDetector/xml/OpenDataDetector.xml --output-json --mat-output-allmaterial true --mat-output-sensitives false
```
Comment (paulgessinger) [inline docs/howto/run_material_mapping.rst:18]: Ah yeah, good catch. I missed that.

**Review by paulgessinger — APPROVED** [review]
Small number of comments, nothing substantial though. LGTM.

**Comment by Corentin-Allaire** [comment]
> Thanks for the comment @paulgessinger, I will have a look. I have one question, right now the CI fail because in json.hpp we get an implicit-fallthrough error. Apparently the default of a switch doesn't have a break statement, technically it does need one because it is the last case but still this make the CI fail. Do you think there is a way around that ?

Actually I had a look and disabling NDEBUG in the CI this would probably work. Not sure if we want to do it though.

**Comment by paulgessinger** [comment]
I guess the preferrable workaround would be to include JSON while ignoring warnings from it. We do this for boost and eigen via `-isystem`, but I don't think we do this for JSON.

**Comment by Corentin-Allaire** [comment]
> I guess the preferrable workaround would be to include JSON while ignoring warnings from it. We do this for boost and eigen via `-isystem`, but I don't think we do this for JSON.
> 
> Edit: maybe we're actually not doing this anymore...
> I guess the next best thing, since there's only **one** JSON header that we include, would be to have a dedicated ACTS JSON header that includes that header, and wraps it in `#pragma`s to suppress this warning.

Something like : 
`#pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
#include <nlohmann/json.hpp>` ?

And then we include this instead of  json.hpp ? Would that cause issue when using an external json ?

**Comment by paulgessinger** [comment]
We should pop the pragmas after the include I guess. I think this shouldn't interact with external JSON either, since the inner include just resolves to what would otherwise be included. Or am I missing something?

**Comment by Corentin-Allaire** [comment]
> We should pop the pragmas after the include I guess. I think this shouldn't interact with external JSON either, since the inner include just resolves to what would otherwise be included. Or am I missing something?

No I think you are right, I am always a bit uneasy when it comes to compilation so I wanted to ask just to be sure. So let me do that and see what happen.

**Review by paulgessinger — APPROVED** [review]
Good to go.

**Pull request merged**